src/platforms/hosted/windows/meson.build

```
@@ -0,0 +1,46 @@
+# This file is part of the Black Magic Debug project.
+#
+# Copyright (C) 2023 1BitSquared <info@1bitsquared.com>
+# Written by Rachel Mant <git@dragonmux.network>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+bmda_sources += files('ftdi.c')
+
+bmda_includes += include_directories('@0@/3rdparty/ftdi'.format(meson.project_source_root()))
+
+bmda_link_args += [
+	cc.find_library(
+		'ftd2xx',
+		dirs: '@0@/3rdparty/ftdi/@1@'.format(meson.project_source_root(), build_machine.cpu())
+	)
+]
+
+configure_file(
+	input: '@0@/3rdparty/ftdi/@1@/ftd2xx.dll'.format(meson.project_source_root(), build_machine.cpu()),
+	output: '@0@/src/ftd2xx.dll'.format(meson.project_build_root()),
+	copy: true,
+)
```

The better syntax for building paths here would be using the `/` [operator](https://mesonbuild.com/Syntax.html#string-path-building)

```suggestion
bmda_includes += include_directories( meson.project_source_root() / '3rdparty/ftdi')

bmda_link_args += [
	cc.find_library(
		'ftd2xx',
		dirs: meson.project_source_root() / '3rdparty/ftdi' / build_machine.cpu()
	)
]

configure_file(
	input: meson.project_source_root() / '3rdparty/ftdi' / build_machine.cpu() / 'ftd2xx.dll'
	output: meson.project_build_root() / 'src/ftd2xx.dll',
	copy: true,
)
```

---

Ah, yes! good call, especially as that eliminates the .format()'ing. We'll adjust that in a short bit and push